### PR TITLE
Add Strife PC speaker sound support

### PR DESCRIPTION
**Pull request by ceski-1**
Fixes https://github.com/chocolate-doom/chocolate-doom/issues/1594.

### CODE REVIEW

**Comment by ceski-1** [comment]
Here's some documentation.

**Overview:**

PC speaker sound support in Strife is functional with some caveats. The IWAD only contains 21 `DP*` sounds even though there are 135 `DS*` sounds to be accounted for. For comparison, Doom has a PC speaker sound for each PCM sound. If one of the sounds is missing, the game will exit with a `W_GetNumForName()` error. So how does Strife avoid this? It uses a strange workaround.

All missing PC speaker sounds are treated as if they were `sfx_rifle` which uses the `dprifle` sound lump. When `I_StartSound()` is called, the game is supposed to check if the requested sound is `sfx_rifle`. If so, it's supposed to skip playback. However, in practice the comparison logic is flawed so every missing sound ends up playing anyway, using `dprifle`.

Ironically, the only comparison that works ends up silencing the assault rifle itself. That's right, it can't use the sound assigned to it. This behavior can be retained for accuracy, but it seems terrible from a user perspective. Currently, every missing PC speaker sound except the assault rifle can be substituted via a PWAD and they will play correctly, as long as the names match their `DS*` counterparts.

**Evaluating strife1.exe 1.31:**

1. Type `boomstix` then fire each weapon, navigate the menus, etc. Note that only the assault rifle is silent, and `sfx_rifle` (`dprifle`) is used in place of all missing sounds.

2. Change `0x05` to `0x03` at offset `0x59595` to change `sfx_rifle` to `sfx_mtalht` (the lumps are identical) in `A_FireRifle()`. Now the assault rifle sound plays correctly.

3. Change `0x01` to `0x00` at offset `0x43509` to bypass "ignore sfx_rifle playback" in `I_StartSound()`. Now the assault rifle sound plays correctly. Other sounds play as well.

4. Change `0x75` to `0x74` at offset `0x43512` to change "ignore sfx_rifle playback" in `I_StartSound()` to "only play the assault rifle sound". All other sounds are ignored, including missing ones.

And so on, but the above is enough to identify the behavior.

**Comment by fragglet** [comment]
Is there any reason this can't just be done with a PWAD that replaces / adds the appropriate lumps? That would presumably work with vanilla Strife too.

**Comment by ceski-1** [comment]
> Is there any reason this can't just be done with a PWAD that replaces / adds the appropriate lumps? That would presumably work with vanilla Strife too.

No. The accurate behavior of Strife, most likely due to an oversight, is to never play a sound for `sfx_rifle`. I would argue that an exception to 100% accuracy should be made and allow this sound to be played. All that means is removing [this section](https://github.com/chocolate-doom/chocolate-doom/pull/1632/files#diff-8b40c349e820839317a9a3f9bbefcbfcb750043c668ca7ca854eef044bbcc98dR187-R193).

**Comment by ceski-1** [comment]
Here's the summary:

Intended (likely) vanilla Strife behavior:
Missing sounds replaced with `sfx_rifle`. When there's an attempt to play `sfx_rifle`, skip it.

Actual vanilla Strife behavior:
Missing sounds replaced with `sfx_rifle` and play normally. But when there's an attempt to play `sfx_rifle` for the assault rifle, skip it.

Proposed behavior for Chocolate Strife:
Missing sounds replaced with `sfx_rifle` and play normally, including the assault rifle. Alternatively, missing sounds can be skipped, but allow the assault rifle to play.

**Review by fragglet — CHANGES_REQUESTED** [review]

**Comment by ceski-1** [comment]
That pretty much sums it up.

> Fixing (3) does make me uncomfortable because it goes against the project philosophy. It seems like such a minor thing that leaving the vanilla behavior and just not playing that sound isn't that huge a deal. What do you think?

Sure, it's probably more appropriate to do this for Crispy Strife.

File: src/i_pcsound.c [hunk 1]
```diff
@@ -244,9 +251,19 @@ static int I_PCS_GetSfxLumpNum(sfxinfo_t* sfx)
...
-    if (use_sfx_prefix)
+    if (gamemission == doom)
+    {
+        M_snprintf(namebuf, sizeof(namebuf), "dp%s", DEH_String(sfx->name));
+    }
+    else if (gamemission == strife)
     {
         M_snprintf(namebuf, sizeof(namebuf), "dp%s", DEH_String(sfx->name));
+
+        if (W_CheckNumForName(namebuf) == -1)
```
[the remark this answers is not in the document]
Comment (ceski-1) [reply src/i_pcsound.c:256]: I think this is cleaner, so I'll go for it.

**Comment by ceski-1** [comment]
@fragglet Everything look good now?

**Review by fragglet — APPROVED** [review]

**Pull request merged**